### PR TITLE
Removed tip suggesting the use of 2 Spack instances

### PR DESCRIPTION
**Pull request by alalazo**
This PR removes a tip in the documentation that is in my opinion controversial. As it is just my opinion though, feel free to close this without mercy if the majority thinks differently.

### CODE REVIEW

**Comment by adamjstewart** [comment]
I also disagree with this tip. I use multiple Spack instances for other reasons, but not this one.

**Comment by citibeth** [comment]
I wrote the tip becuase I do use it.  And it's saved me a lot of pain not having to re-build GCC and then change `compilers.yaml` for really no good reason.  If you don't like the tip, don't use it, otherwise what problem does this create?

**Comment by tgamblin** [comment]
I vote to remove bc this confuses new users.  We should try to keep the docs simple, and this complicates setup and forces people to remember which spack they're using in what context, when one would do fine.

**Comment by svenevs** [comment]
Please merge this, it causes more confusion than it does help.  I want to be clear that it generally is **good** advice, but not in its current form.  An "advanced" docs section would be where this could go, _provided that_ how to navigate a dual-spack (patent pending) installation receive significantly more explanation as to _how_ to do this.

**I STILL AGREE WITH THE ADVICE** even after the battle of learning how to finesse this setup.  But heaven help the poor soul who has to maintain this cluster in the future.  Simply stating "use this one for compilers and core utilities" means anybody who does it has to re-learn how to coordinate the two.  And if they want to use `lmod`, it gets very complicated.

- It's very nice to feel "safe" that my compilers are done and won't be messed with.
- I get to have different `packages.yaml` for each, which in this case was very helpful (e.g. what the default compiler for `all` is).
- It's annoying that anything we decide to re-whitelist, or blacklist, or whatever, requires full module regeneration and inevitably direct modification (manual, or swanky patch-based hacking that is going to be irrelevant if we ever update).
- The documentation related to this **must** explain how to setup the other spack instance with the appropriate `packages.yaml`.
    - For example, I stupidly forgot to `buildable: False` for `cmake` and point to the other one, and now it's way too late to undo that.  Same for `curl`.
        - So now our users `module switch` and will probably be very confused as to why their `cmake` and `curl` are getting switched when they switch compilers.
- The documentation related to this must also explain how a user should configure the site-level (`etc/spack`) `compilers.yaml` to use the "core" spack compilers.

Getting this cluster setup was an adventure, and some of the things I had to do were simply because I had already compiled two versions of GCC and LLVM, and we literally don't have time to redo that.

In short, I wouldn't say this is "bad advice" because it really does provide you with some nice safety features.  But it needs an entire article on its own to guide users as to how this should be executed.  @citibeth I'm more than happy to help write that article with you, but particularly for new users the documentation can't guide them toward this without explaining the implications.
